torch/csrc/generic/StorageSharing.cpp

```diff
@@ -212,27 +212,49 @@ static PyObject * THPStorage_(newSharedFd)(PyObject *_unused, PyObject *args)
 
 #else // THC_GENERIC_FILE
 
-static PyObject * THPStorage_(shareCuda)(THPStorage *self)
-{
+static PyObject* THPStorage_(shareCuda)(THPStorage* self) {
   HANDLE_TH_ERRORS
-  THWStorage *storage = self->cdata;
+  THWStorage* storage = self->cdata;
+
+  if (storage->received_cuda()) {
+    AT_ERROR(
+        "Attempting to send tensor received from another process. Consider clonning.");
```

Comment:
We can make this error more legible to users by saying exactly why this doesn't work. "Attempted to send tensor received from another process; this is not currently supported. Consider cloning before sending." Also, typo: clonning